crates/ekore/src/constants.rs

```diff
@@ -41,3 +52,21 @@ pub const PID_NSM: u16 = 10201;
 
 /// non-singlet all-valence |PID|.
 pub const PID_NSV: u16 = 10200;
+
+/// compute the number of up flavors
+pub fn uplike_flavors(nf: u8) -> u8 {
+    if nf > 6 {
+        unimplemented!("Selected nf is not implemented")
+    }
+    nf / 2
+}
+
+pub fn charge_combinations(nf: u8) -> Vec<f64> {
+    let nu = uplike_flavors(nf) as f64;
+    let nd = (nf as f64) - nu;
+    let e2avg = (nu * eu2 + nd * ed2) / (nf as f64);
+    let vue2m = nu / (nf as f64) * (eu2 - ed2);
+    let vde2m = nd / (nf as f64) * (eu2 - ed2);
+    let e2delta = vde2m - vue2m + e2avg;
+    vec![e2avg, vue2m, vde2m, e2delta]
```

Comment:
Should we return a dedicate `struct`? We could make `nu` and `nd` part of that ...

Reply:
yes for sure. I was going to add a comment asking for input on how to code this bit properly. I ll add a dedicate `struct`